ddtrace/http/headers.py

```diff
@@ -0,0 +1,86 @@
+import logging
+import re
+from ..utils.http import normalize_header_name
+
+log = logging.getLogger(__name__)
+
+REQUEST = 'request'
+RESPONSE = 'response'
+
+NORMALIZE_PATTERN = re.compile(r'([^a-z0-9])+')
+
+
+def store_request_headers(headers, span, integration_config):
+    """
+    Store request headers as a span's tags
+    :param headers: All the request's http headers, will be filtered through the whitelist
+    :type headers: dict or list
+    :param span: The Span instance where tags will be stored
+    :type span: ddtrace.Span
+    :param integration_config: An integration specific config object.
+    :type integration_config: ddtrace.settings.IntegrationConfig
+    """
+    _store_headers(headers, span, integration_config, REQUEST)
+
+
+def store_response_headers(headers, span, integration_config):
+    """
+    Store response headers as a span's tags
+    :param headers: All the response's http headers, will be filtered through the whitelist
+    :type headers: dict or list
+    :param span: The Span instance where tags will be stored
+    :type span: ddtrace.Span
+    :param integration_config: An integration specific config object.
+    :type integration_config: ddtrace.settings.IntegrationConfig
+    """
+    _store_headers(headers, span, integration_config, RESPONSE)
+
+
+def _store_headers(headers, span, integration_config, request_or_response):
+    """
+    :param headers: A dict of http headers to be stored in the span
+    :type headers: dict or list
+    :param span: The Span instance where tags will be stored
+    :type span: ddtrace.span.Span
+    :param integration_config: An integration specific config object.
+    :type integration_config: ddtrace.settings.IntegrationConfig
+    """
+    if not isinstance(headers, dict):
+        try:
+            headers = dict(headers)
+        except:
```

Comment:
We do not want a bare except here, we want `except TypeError`, or if we want to be safer, `except Exception`

Reply:
The reason for this is that some exceptions do not inherit from `Exception`, e.g. `KeyboardInterrupt`. So if you have a bare except and someone did `ctrl+c` while in this try/except, then it would not bubble up and stop execution of the program.